src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/player/ModuleEagle.kt

```diff
@@ -92,7 +100,14 @@ object ModuleEagle : ClientModule("Eagle", Category.PLAYER,
     ) { event ->
         val shouldBeActive = !player.abilities.flying && Conditional.shouldSneak(event)
 
-        event.sneak = shouldBeActive && player.isCloseToEdge(event.directionalInput, edgeDistance.toDouble())
+        event.sneak = shouldBeActive && player.isCloseToEdge(event.directionalInput, currentEdgeDistance.toDouble())
+    }
+
+    val tickHandler = tickHandler {
+        waitTicks(edgeDistanceResetTime.random())
+        if (player.moving) {
+            currentEdgeDistance = edgeDistance.random()
```

Comment:
That could cause very bad behaviour, basically causing it to sneak/unsneak spam on move. Current Edge Distance should be randomized **on unsneak** (not if is not sneaking, but was sneaking and is not sneaking) instead, that would also eliminate the requirement for edgeDistanceResetTime